executor.go

```diff
@@ -272,66 +277,65 @@ type executeFieldsParams struct {
 }
 
 // Implements the "Evaluating selection sets" section of the spec for "write" mode.
-func executeFieldsSerially(p executeFieldsParams) *Result {
+func executeFieldsSerially(p executeFieldsParams, resultPool ResultPool) *Result {
 	if p.Source == nil {
 		p.Source = map[string]interface{}{}
 	}
 	if p.Fields == nil {
 		p.Fields = map[string][]*ast.Field{}
 	}
 
-	finalResults := make(map[string]interface{}, len(p.Fields))
+	result := resultPool.Get()
+	object := resultPool.GetObjectFor(result, len(p.Fields))
 	for _, orderedField := range orderedFields(p.Fields) {
 		responseName := orderedField.responseName
 		fieldASTs := orderedField.fieldASTs
-		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs)
+		resolved, state := resolveField(p.ExecutionContext, p.ParentType, p.Source, fieldASTs, result, resultPool)
 		if state.hasNoFieldDefs {
 			continue
 		}
-		finalResults[responseName] = resolved
+		object[responseName] = resolved
 	}
-	dethunkMapDepthFirst(finalResults)
+	dethunkMapDepthFirst(object)
```

Comment:
Can we leave the original variables  names the same as much as possible? Merging master will be very challenging if we change too much.

```suggestion
	dethunkMapDepthFirst(finalResults)
```

Reply:
done.